evaluator/evaluator.py

```diff
@@ -82,26 +82,32 @@ def _load_cves_for_inventory_id(system_id, cur):
 
     def _register_missing_cves(self, missing_cves, cur):
         cve_id_map = {}
+        new_cves = set()
         for cve in missing_cves:
             to_insert = ((cve, 'unknown', 0,), cve,)
             # do nothing if row exists - other evaluator or vmaas-sync just inserted it
             cur.execute("""INSERT INTO cve_metadata
                            (cve, description, impact_id)
                            VALUES %s ON CONFLICT (cve) DO UPDATE set cve = %s
                            returning id""", to_insert)
-            cve_id_map[cve] = cur.fetchone()[0]
+            cve_id = cur.fetchone()[0]
+            cve_id_map[cve] = cve_id
+            new_cves.add(cve_id)
         self._cve_id_map.update(cve_id_map)
+        return new_cves
 
     def _store_new_cves(self, system_id, inventory_id, new_cves, rh_account, cur):
+        new_cves_ids = set()
         if not new_cves:
-            return
+            return new_cves_ids
         cvss_buckets = {'0to3': 0, '3to7': 0, '7to10': 0}
 
         # FIXME: getting metadata from DB on every system update should be optimized
         cur.execute("select cve, coalesce(cvss3_score, cvss2_score) from cve_metadata where cve in %s",
                     (tuple(new_cves),))
         cves_in_db = set()
         for cve_tuple in cur.fetchall():
+            new_cves_ids.add(self._cve_id_map[cve_tuple[0]])
```

Comment:
this might fail with KeyError when cve is selected from DB but is not in map yet

Reply:
Should be fixed now.